File: wiki-java/src/main/java/dev/freya02/wiki/autocomplete/transformer/FullNameTransformer.java
package dev.freya02.wiki.autocomplete.transformer;

import io.github.freya022.botcommands.api.commands.application.slash.autocomplete.AutocompleteTransformer;
import io.github.freya022.botcommands.api.core.service.annotations.BService;
import net.dv8tion.jda.api.interactions.commands.Command;
import org.jspecify.annotations.NullMarked;

// --8<-- [start:autocomplete_transformer-java]
@BService
@NullMarked // Everything is non-null unless @Nullable
public class FullNameTransformer implements AutocompleteTransformer<FullName> {
    @Override
    public Class<FullName> getElementType() {
        return FullName.class;
    }

    @Override
    public Command.Choice apply(FullName fullName) {
        return new Command.Choice(
                "%s %s".formatted(fullName.firstName(), fullName.secondName()),
                "%s|%s".formatted(fullName, fullName.secondName())
        );
    }
}
// --8<-- [end:autocomplete_transformer-java]

File: wiki-kotlin/src/main/kotlin/io/github/freya022/wiki/autocomplete/transformer/FullNameTransformer.kt
package io.github.freya022.wiki.autocomplete.transformer

import io.github.freya022.botcommands.api.commands.application.slash.autocomplete.AutocompleteTransformer
import io.github.freya022.botcommands.api.core.service.annotations.BService
import net.dv8tion.jda.api.interactions.commands.Command

// --8<-- [start:full_name_obj-kotlin]
data class FullName(val firstName: String, val secondName: String)
// --8<-- [end:full_name_obj-kotlin]

// --8<-- [start:autocomplete_transformer-kotlin]
@BService
class FullNameTransformer : AutocompleteTransformer<FullName> {
    override val elementType: Class<FullName> = FullName::class.java

    override fun apply(e: FullName): Command.Choice {
        return Command.Choice("${e.firstName} ${e.secondName}", "${e.firstName}|${e.secondName}")
    }
}
// --8<-- [end:autocomplete_transformer-kotlin]
